metomi/rosie/cli_utils.py

```diff
@@ -0,0 +1,179 @@
+# Copyright (C) British Crown (Met Office) & Contributors.
+# This file is part of Rose, a framework for meteorological suites.
+#
+# Rose is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Rose is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Rose. If not, see <http://www.gnu.org/licenses/>.
+# -----------------------------------------------------------------------------
+
+from subprocess import Popen, PIPE
+import textwrap
+from typing import Optional, List
+
+# NOTE: Box Drawing characters
+# t, b, l, r -> top, bottom, left, right
+# h, v -> horizontal, vertical
+# l, x -> line, cross (i.e. vertex)
+
+# unicode (U_) chars
```

Comment:
Heavily inspired by the Cylc `--box` format system.

Search the internet for "box drawing characters" for more info.